osu.Game/Online/Chat/MessageFormatter.cs

```diff
@@ -20,7 +20,7 @@ public static class MessageFormatter
         private static readonly Regex new_link_regex = new Regex(@"\[(?<url>[a-z]+://[^ ]+) (?<text>(((?<=\\)[\[\]])|[^\[\]])*(((?<open>\[)(((?<=\\)[\[\]])|[^\[\]])*)+((?<close-open>\])(((?<=\\)[\[\]])|[^\[\]])*)+)*(?(open)(?!)))\]");
 
         // [test](https://osu.ppy.sh/b/1234) -> test (https://osu.ppy.sh/b/1234) aka correct markdown format
```

Comment:
Please provide a more detailed description of what the regex should do.

not everyone is able to understand regex right away.

Reply:
I'm not sure what I could explain further here without restating the regex verbatim, part-by-part, which seems counterproductive - this does not seem like the correct place to explain regular expression syntax. (And if you check the commit message, I explained the added fragment part-by-part - the explanation is in the PR description, too).